frontend/src/pages/Player/Toolbar/DevToolsWindowV2/ConsolePage/ConsolePage.tsx

```diff
@@ -1,96 +1,92 @@
-import { Virtuoso, VirtuosoHandle } from 'react-virtuoso'
-
-import { TIME_FORMAT } from '@/components/Search/SearchForm/constants'
-import { useGetSessionLogsQuery } from '@/graph/generated/hooks'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useGetLogs } from '@/pages/LogsPage/useGetLogs'
+import { LogCustomColumn } from '@/components/CustomColumnPopover'
 import { buildSessionParams } from '@/pages/LogsPage/utils'
-import { styledVerticalScrollbar } from '@/style/common.css'
 import analytics from '@/util/analytics'
+import { parseSearch } from '@/components/Search/utils'
+import { DEFAULT_LOG_COLUMNS } from '@/pages/LogsPage/LogsTable/CustomColumns/columns'
 
 import { useReplayerContext } from '../../../ReplayerContext'
-import * as styles from './style.css'
-
-type SessionLog = Pick<Log, 'timestamp' | 'message' | 'level'>
-type SessionLogEdge = { cursor: string; node: SessionLog }
+import { ConsoleTable } from '../ConsoleTable'
 
 export const ConsolePage = ({
-	logCursor,
 	autoScroll,
-	filter,
-	sources,
-	levels,
+	panelHeight,
+	query,
 }: {
 	autoScroll: boolean
 	logCursor: string | null
 	filter: string
-	sources: LogSource[]
-	levels: LogLevel[]
+	panelHeight: number
+	query: string
 }) => {
 	const { projectId } = useProjectId()
-	const [, setSelectedCursor] = useState(logCursor)
-	const { session, time, setTime, sessionMetadata, isPlayerReady } =
-		useReplayerContext()
-
-	const params = buildSessionParams({ session, levels, sources })
-
-	const { data, loading } = useGetSessionLogsQuery({
-		variables: {
-			params: {
-				query: params.query,
-				date_range: {
-					start_date:
-						params.date_range.start_date.format(TIME_FORMAT),
-					end_date: params.date_range.end_date.format(TIME_FORMAT),
-				},
-			},
-			project_id: projectId,
-		},
-		skip: !session,
+	const { session, time, setTime, sessionMetadata } = useReplayerContext()
+
+	const params = buildSessionParams({ session, query })
+	const { queryParts } = parseSearch(params.query)
+
+	const {
+		logEdges,
+		loading,
+		error,
+		loadingAfter,
+		fetchMoreForward,
+		refetch,
+	} = useGetLogs({
+		query: params.query,
+		project_id: projectId,
+		startDate: params.date_range.start_date.toDate(),
+		endDate: params.date_range.end_date.toDate(),
+		disablePolling: true,
+		disableRelatedResources: true,
+		logCursor: undefined,
+		sortDirection: SortDirection.Asc,
+		sortColumn: 'timestamp',
+		limit: 1_000,
 	})
 
-	const messagesToRender = useMemo(() => {
-		if (!data?.sessionLogs) {
-			return []
-		}
+	const fetchMoreWhenScrolled = useCallback(
+		(containerRefElement?: HTMLDivElement | null) => {
+			if (containerRefElement) {
+				const { scrollHeight, scrollTop, clientHeight } =
+					containerRefElement
 
-		if (filter !== '') {
-			return data.sessionLogs.filter((logEdge) => {
-				if (!logEdge.node.message) {
-					return false
+				if (scrollHeight - scrollTop - clientHeight < 100) {
+					fetchMoreForward()
 				}
```

Comment:
makes sense, and with the logs results being driven by the query, searching for something that matches results towards the end of the logs would be good since the search would be server side. something to implement for the network request trace search with the new table